src/node_process_object.cc

```diff
@@ -105,18 +105,37 @@ MaybeLocal<Object> CreateProcessObject(Realm* realm) {
   READONLY_PROPERTY(
       process, "version", FIXED_ONE_BYTE_STRING(isolate, NODE_VERSION));
 
-  // process.versions
   Local<Object> versions = Object::New(isolate);
-  READONLY_PROPERTY(process, "versions", versions);
+  // Node.js version is always on the top
+  READONLY_STRING_PROPERTY(
+      versions, "node", per_process::metadata.versions.node);
+  std::vector<std::pair<std::string_view, std::string_view>> versions_vec;
 
 #define V(key)                                                                 \
-  if (!per_process::metadata.versions.key.empty()) {                           \
-    READONLY_STRING_PROPERTY(                                                  \
-        versions, #key, per_process::metadata.versions.key);                   \
-  }
+  do {                                                                         \
+    versions_vec.push_back(                                                    \
+        std::make_pair(#key, per_process::metadata.versions.key));             \
+  } while (0);
   NODE_VERSIONS_KEYS(V)
 #undef V
 
+  std::sort(versions_vec.begin(), versions_vec.end(), [](auto& a, auto& b) {
+    return a.first < b.first;
+  });
+
+  for (const auto& version : versions_vec) {
+    versions
+        ->DefineOwnProperty(
+            context,
+            OneByteString(isolate, version.first.begin(), version.first.size()),
```

Comment:
actually, sorry for catching this late:

```suggestion
            OneByteString(isolate, version.first.data(), version.first.size()),
```

since `.begin()` returns an iterator, which isn’t necessarily a `const char*`, e.g. in Github Actions CI:

```
D:\a\node\node\src\node_process_object.cc(130,13): error C2665: 'node::OneByteString': none of the 3 overloads could convert all the argument types [D:\a\node\node\libnode.vcxproj]
D:\a\node\node\src\util-inl.h(208,30): message : could be 'v8::Local<v8::String> node::OneByteString(v8::Isolate *,const unsigned char *,int)' [D:\a\node\node\libnode.vcxproj]
D:\a\node\node\src\util-inl.h(199,30): message : or       'v8::Local<v8::String> node::OneByteString(v8::Isolate *,const signed char *,int)' [D:\a\node\node\libnode.vcxproj]
D:\a\node\node\src\util-inl.h(190,30): message : or       'v8::Local<v8::String> node::OneByteString(v8::Isolate *,const char *,int)' [D:\a\node\node\libnode.vcxproj]
D:\a\node\node\src\node_process_object.cc(131,69): message : while trying to match the argument list '(v8::Isolate *, std::_String_view_iterator<_Traits>, std::basic_string_view<char,std::char_traits<char>>::size_type)' [D:\a\node\node\libnode.vcxproj]
```